packages/react-ui/src/theme/carto-theme.d.ts

```diff
@@ -8,17 +8,68 @@ import { Palette, PaletteColor } from '@mui/material'
 */
 export const cartoThemeOptions: ThemeOptions;
 
-type Modify<T, R> = Omit<T, keyof R> & R
-type CustomPaletteColor = Modify<PaletteColor, { contrastText: string, relatedLight: string, background: string }>
-type CustomPalette = Modify<Palette, {
-  primary: CustomPaletteColor,
-  secondary: CustomPaletteColor,
-  error: CustomPaletteColor,
-  warning: CustomPaletteColor,
-  info: CustomPaletteColor,
-  success: CustomPaletteColor,
-}>
+type Modify<T, R> = Omit<T, keyof R> & R;
+type CustomPaletteColor = Modify<
+  PaletteColor,
+  {
+    relatedDark: string;
+    relatedLight: string;
+    background: string;
+    outlinedBorder: string;
+    hint: string;
+    navyBlue: string;
+    locationRed: string;
+    predictionBlue: string;
+    softBlue: string;
```

Comment:
Are different palettes gonna have variations of these props? It feels a bit strange. I don't want to block it but we should discuss about it

Reply:
No, I added it this way to simplify, but you're right. I've just updated this file to be more granular. Let me know if it's ok.